lib/ConfigLib.sol

```diff
@@ -3,214 +3,95 @@ pragma solidity ^0.8.17;
+        MultisigIsm.DomainConfig[]
+            memory domains = new MultisigIsm.DomainConfig[](chainNames.length);
         for (uint256 i = 0; i < chainNames.length; i++) {
-            string memory chainName = chainNames[i];
-            domains[i] = readMultisigIsmDomainConfig(vm, chainName);
+            domains[i] = readMultisigIsmDomainConfig(vm, chainNames[i]);
         }
-        return MultisigIsmConfig(domains);
+        return Multisig(domains, owner);
     }
 
-    function writeAgentConfig(
-        HyperlaneDomainConfig memory config,
-        Vm vm,
-        uint256 startBlock
-    ) internal {
-        string memory baseConfig = "config";
-        vm.serializeString(
-            baseConfig,
-            "domain",
-            vm.toString(uint256(config.domainId))
-        );
-        vm.serializeString(baseConfig, "rpcStyle", "ethereum");
-        vm.serializeString(baseConfig, "finalityBlocks", "POPULATE_ME");
-
-        string memory addresses = "addresses";
-        vm.serializeAddress(addresses, "mailbox", address(config.mailbox));
-        vm.serializeString(
-            baseConfig,
-            "addresses",
-            vm.serializeAddress(
-                addresses,
-                "interchainGasPaymaster",
-                address(config.igp)
-            )
-        );
-
-        string memory connection = "connection";
-        vm.serializeString(connection, "type", "http");
-        vm.serializeString(
-            baseConfig,
-            "connection",
-            vm.serializeString(connection, "url", "")
-        );
-
-        string memory index = "index";
-        vm.serializeString(
-            baseConfig,
-            "index",
-            vm.serializeString(index, "from", vm.toString(startBlock))
-        );
-
-        vm.serializeString(baseConfig, "name", config.chainName).write(
-            string.concat("./config/", config.chainName, "_agent_config.json")
-        );
+    function digest(Multisig memory multisig) internal pure returns (bytes32) {
+        return keccak256(abi.encode(multisig));
     }
 
-    function write(HyperlaneDomainConfig memory config, Vm vm) internal {
-        string memory contracts = "contracts";
-        vm.serializeAddress(contracts, "mailbox", address(config.mailbox));
-        vm.serializeAddress(
-            contracts,
-            "interchainGasPaymaster",
-            address(config.igp)
-        );
-        vm.serializeAddress(contracts, "proxyAdmin", address(config.admin));
-        vm.serializeAddress(
-            contracts,
-            "testRecipient",
-            address(config.testRecipient)
-        );
-        vm
-            .serializeAddress(
-                contracts,
-                "create2Factory",
-                address(config.create2)
-            )
-            .write(
-                "./config/networks.json",
-                string.concat(".", config.chainName, ".contracts")
-            );
-    }
+    // function buildAgentConfig(
```

Comment:
What do we need to do to get this back? It's useful (for now) for folks spinning up agents